packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -147,9 +147,12 @@ export class BlobManager {
     }
 
     public processBlobAttachOp(blobId: string, local: boolean) {
-        assert(!local || this.pendingBlobIds.has(blobId), 0x1f8 /* "local BlobAttach op with no pending blob" */);
-        this.pendingBlobIds.get(blobId)?.resolve();
-        this.pendingBlobIds.delete(blobId);
+        if (local) {
```

Comment:
Your code looks correct, but I wonder why you chose this design over using first op?
I.e. I think you can leave the code as is, but remove assert. We lose a bit of validation that way.
Either way, it's probably does not matter :)

Curios, how did we find this bug in the first place?

Reply:
It seemed simpler this way and I would guess it's possible to hit this in normal usage but very unlikely so like you say it probably doesn't matter

Reply:
issue was discovered by writing a fuzz tester for the tree DDS whiteboard uses ahead of shipping the op/summary compression work. The 0.1.1 format with history involves a strategy for chunking edits into blobs which are uploaded. There's assumption that all clients agree on the chunk boundaries which is partially enforced via quorum, but in certain scenarios chunks can get uploaded twice. For example if the leader elected to upload blobs leaves the session immediately after submitting an edit chunk, and the newly elected leader starts uploading yet-un-uploaded blobs before blob attach op gets broadcast to it.

It should actually be a rare scenario in practice.